### ISSUE TITLE
TestSyncAuthorizedEntries appears possibly flaky in ubuntu CI

### ISSUE THREAD

**Issue by zmt** [spiffe/spire#4760]
I noticed a failure in an unrelated PR: https://github.com/spiffe/spire/actions/runs/7286242421/job/19854617633?pr=4747. I am unable to repro locally on a linux system (debian 10.11):
```
% go test -run=TestSyncAuthorizedEntries/fetcher_fails -failfast -count=10000 .
ok  	github.com/spiffe/spire/pkg/server/api/entry/v1	66.890s
```

* **Version**: `main`
* **Platform**: `ubuntu 20.04`
* **Subsystem**: server

**Comment by zmt** [comment]
I think this was closed by #4769

**Comment by azdagron** [comment]
Yes, this is fixed now.
